script/script.js

```diff
@@ -0,0 +1,174 @@
+            par.textContent = text;
+            this.inkVolume -= str.length/2;
+            console.log(`Your ink volume is ${this.inkVolume}.`)
+        } else {
+            console.log(`The amount of ink in the marker is not enough to write the whole text.`); 
+            let substr = text.substring(0, this.inkVolume*2);
+            str = substr.replace(/\s+/g, '');
+            let numberOfSpace = substr.length - str.length;
+            if (numberOfSpace == 0) {
+                par.textContent = substr;
+            } else {
+                substr = text.substring(0, this.inkVolume*2 + numberOfSpace);
+                par.textContent = substr + `...`;
+            }
+            alert("The marker ink is out.")
+            this.inkVolume = 0;
+        }
+    }
+}
+
+let redMarker = new Marker("red");
+redMarker.inputTextAndPrint();
+
+class RefillMarker extends Marker {
+    constructor (color) {
+        super(color);
+        this.inkVolume = 100;
+    }
+
+    inputTextAndPrint () {
+        super.inputTextAndPrint();
+    }
+
+    refill () {
+        this.inkVolume = 100;
+        console.log(`Your ink volume is already full.`)
+    }
+}
+
+let blueMarker = new RefillMarker("blue");
+blueMarker.inputTextAndPrint();
+blueMarker.inputTextAndPrint();
+blueMarker.refill();
+blueMarker.inputTextAndPrint();
+
+// 3. Реалізуй клас Employee, що описує працівника, і створи масив працівників банку.
+// Реалізуй клас EmpTable для генерації HTML-коду таблиці зі списком працівників банку.
+// Масив працівників необхідно передавати через конструктор, а отримувати HTML-код за допомогою методу getHtml().
+// Створи об'єкт класу EmpTable і виведи на екран результат роботи методу getHtml().
+
+class Employee {
+    constructor (firstname, lastname, age, position) {
+        this.firstname = firstname;
+        this.lastname = lastname;
+        this.age = age;
+        this.position = position;
+    }
+}
+
+let employee_1 = new Employee("Mariia", "Tyshkovets", 26, "engineer");
+let employee_2 = new Employee("Alla", "Mytrophanova", 27, "accountant");
+let employee_3 = new Employee("Andrew", "Tyshkovets", 27, "programmer");
+let employee_4 = new Employee("Semen", "Malyy", 15, "cleaner");
+let employee_5 = new Employee("Petro", "Gorbatyy", 70, "secretary");
+
+
+let employees = [employee_1, employee_2, employee_3, employee_4, employee_5];
+
+class EmpTable {
+    constructor (array) {
+        let newArray = [];
+        newArray = array.map(a => {
+            let employee = {};
+            Object.keys(a).forEach(propertyKey => {
+                employee[propertyKey] = a[propertyKey];
+            });
+            return employee;
+        });
+    }
+
+    getHTML () {
+        let tbody = document.getElementsByClassName("tbody")[0];
+        for (let i = 0; i < employees.length; i++) {
+            let tr = document.createElement("TR");
+            tbody.appendChild(tr);
+            for (const property in employees[i]) {
```

Comment:
Be careful with `for...in` loop for objects. It will loop through object's prototypes props as well